src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Source/AnalyzeDiagnostics.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.DotNet.UpgradeAssistant.Analysis;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Source
+{
+    public class AnalyzeDiagnostics : IAnalyzeResultProvider
+    {
+        private readonly IEnumerable<DiagnosticAnalyzer> _allAnalyzers;
+        private readonly ImmutableArray<AdditionalText> _additionalTexts;
+        private readonly ILogger _logger;
+
+        public string Name => "API Upgradability";
+
+        public Uri InformationURI => new("https://docs.microsoft.com/en-us/dotnet/core/porting/upgrade-assistant-overview");
+
+        public AnalyzeDiagnostics(IEnumerable<DiagnosticAnalyzer> analyzers, IEnumerable<AdditionalText> additionalTexts, ILogger<AnalyzeDiagnostics> logger)
```

Comment:
This appears to duplicate a lot of logic from the source updaters. Can a new service be created that provides the diagnostics to minimize the duplication?